### PR TITLE
CB-5566 Upgrade CM to 7.1.4 and remove all the bigdecimal value.

### PR DESCRIPTION
**Pull request by doktoric**
See detailed description in the commit message.

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/ClouderaManagerPollingServiceProvider.java [hunk 1]
```diff
@@ -140,36 +142,36 @@ public PollingResult startPollingCmParcelStatus(Stack stack, ApiClient apiClient
...
-        return pollCommandWithTimeListener(stack, apiClient, BigDecimal.ZERO, POLL_FOR_ONE_HOUR,
+        return pollCommandWithTimeListener(stack, apiClient, ZERO, POLL_FOR_ONE_HOUR,
                 new ClouderaManagerParcelStatusListenerTask(clouderaManagerApiPojoFactory, cloudbreakEventService, parcelVersions, parcelStatus));
     }
 
     public PollingResult startPollingCmParcelDelete(Stack stack, ApiClient apiClient, Map<String, String> parcelVersions) {
         LOGGER.debug("Waiting for Cloudera Manager parcels {} to be deleted. [Server address: {}]", parcelVersions,
                 stack.getClusterManagerIp());
-        return pollCommandWithTimeListener(stack, apiClient, BigDecimal.ZERO, POLL_FOR_ONE_HOUR,
+        return pollCommandWithTimeListener(stack, apiClient, 0, POLL_FOR_ONE_HOUR,
```
Comment (keyki) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/ClouderaManagerPollingServiceProvider.java:152]: The const should be used.

File: gradle.properties [hunk 2]
```diff
@@ -36,7 +36,7 @@ bouncycastleVersion=1.64
 hadoopVersion=2.7.1
 postgreSQLVersion=42.2.5
 httpClientVersion=4.5.10
-cmClientVersion=7.1.1
+cmClientVersion=7.1.4
```
Comment (keyki) [inline gradle.properties:39]: Should we go to a much newer released version? My worry is that if it comes from a private repo they might delete it at some point?

File: datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java [hunk 3]
```diff
@@ -344,7 +344,9 @@ public RangerCloudIdentitySyncStatus setRangerCloudIdentityMapping(String envCrn
     @Override
     @InternalOnly
     public RangerCloudIdentitySyncStatus getRangerCloudIdentitySyncStatus(String envCrn, long commandId) {
-        return rangerCloudIdentityService.getRangerCloudIdentitySyncStatus(envCrn, commandId);
+        return rangerCloudIdentityService.getRangerCloudIdentitySyncStatus(
+                envCrn,
+                Integer.valueOf(String.valueOf(commandId)));
```
Comment (lajosrodek) [inline datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java:349]: Why is this better than `(int) commandId`? Or do you perhaps depend on the `NumberFormatException` thrown by `Integer.valueOf()`?

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/commands/SyncApiCommandRetriever.java [hunk 4]
```diff
@@ -105,16 +105,16 @@ private BigDecimal getLatestCommandId(String commandName, List<CommandResource>
                     .filter(c -> c.getStart() != null)
                     .peek(c -> LOGGER.debug("Found latest {} command with [command_id: {}, start: {}] by {} call", commandName, c.getId(), c.getStart(), path))
                     .max(Comparator.comparingLong(CommandResource::getStart))
-                    .map(c -> new BigDecimal(c.getId()))
+                    .map(c -> Integer.valueOf(String.valueOf(c.getId())))
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/commands/SyncApiCommandRetriever.java:108]: Like in `SdxController.java `, why is this better than `(int) c.getId()`?

File: datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java [hunk 5]
```diff
@@ -180,10 +180,10 @@ public boolean isCloudIdMappingSupported(String stackCrn) throws ApiException {
         }
     }
 
-    public ApiCommand getApiCommand(String stackCrn, long commandId) throws ApiException {
+    public ApiCommand getApiCommand(String stackCrn, int commandId) throws ApiException {
         ApiClient client = clouderaManagerProxiedClientFactory.getProxiedClouderaManagerClient(stackCrn);
         CommandsResourceApi commandsResourceApi = clouderaManagerApiFactory.getCommandsResourceApi(client);
-        return commandsResourceApi.readCommand(BigDecimal.valueOf(commandId));
+        return commandsResourceApi.readCommand(Integer.valueOf(commandId));
```
Comment (lajosrodek) [inline datalake/src/main/java/com/sequenceiq/datalake/cm/ClouderaManagerRangerUtil.java:186]: Why not use autoboxing (i.e. simply `readCommand(commandId)`)?

**Review by lajosrodek — COMMENTED** [review]
2 more unwanted notes. 😄

File: gradle.properties (hunk 2 above)
Comment (doktoric) [reply gradle.properties:39]: would be nice but https://jira.cloudera.com/browse/RELENG-13626

**Comment by lajosrodek** [comment]
Thanks for the fixes. 😄

**Comment by keyki** [comment]
> Thanks for the fixes. 😄

Success story :D

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSetupService.java [hunk 6]
```diff
@@ -270,12 +271,15 @@ public void installCluster(String template) {
                             .importClusterTemplate(calculateAddRepositories(apiClusterTemplate, prewarmed), apiClusterTemplate);
                     ClusterCommand clusterCommand = new ClusterCommand();
                     clusterCommand.setClusterId(cluster.getId());
-                    clusterCommand.setCommandId(apiCommand.getId());
+                    clusterCommand.setCommandId(BigDecimal.valueOf(apiCommand.getId()));
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSetupService.java:274]: Does this still need `BigDecimal` then?

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/ClouderaManagerPollingServiceProvider.java [hunk 7]
```diff
@@ -70,6 +69,8 @@
 
     private static final long POLL_FOR_TWO_HOURS = TimeUnit.HOURS.toSeconds(2);
 
+    private static final int ZERO = 0;
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/ClouderaManagerPollingServiceProvider.java:72]: I love this constant name. 😃

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerCollectDiagnosticsListenerTask.java [hunk 8]
```diff
@@ -69,7 +69,7 @@ private void digForFailureCause(ApiCommand apiCommand, List<String> errorReasons
                 .forEach(cmd -> digForFailureCause(cmd, childErrors, commandsResourceApi));
 
         if (childErrors.isEmpty() && StringUtils.isNotEmpty(apiCommand.getResultMessage())) {
-            String reason = String.format("Command [%s], with id [%.0f] failed: %s", apiCommand.getName(), apiCommand.getId(), apiCommand.getResultMessage());
+            String reason = String.format("Command [%s], with id [%s] failed: %s", apiCommand.getName(), apiCommand.getId(), apiCommand.getResultMessage());
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerCollectDiagnosticsListenerTask.java:72]: Is `apiCommand.getId()` really a `String`?

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandApiErrorParserUtil.java [hunk 9]
```diff
@@ -55,7 +55,7 @@ private ClouderaManagerCommandApiErrorParserUtil() {
                     .forEach(cmd -> errors.addAll(getErrors(cmd, commandsResourceApi)));
 
             if (errors.isEmpty() && StringUtils.isNotEmpty(apiCommand.getResultMessage())) {
-                String reason = String.format("Command [%s], with id [%.0f] failed: %s",
+                String reason = String.format("Command [%s], with id [%s] failed: %s",
```
Comment (lajosrodek) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandApiErrorParserUtil.java:58]: Is `apiCommand.getId()` really a `String`?

**Review by lajosrodek — CHANGES_REQUESTED** [review]
Small asks / questions.

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSetupService.java (hunk 6 above)
Comment (doktoric) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerSetupService.java:274]: unfortunately the CM api is not as consistent than you think :D

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerCollectDiagnosticsListenerTask.java (hunk 8 above)
Comment (doktoric) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerCollectDiagnosticsListenerTask.java:72]: does it matter?

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandApiErrorParserUtil.java (hunk 9 above)
Comment (doktoric) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/util/ClouderaManagerCommandApiErrorParserUtil.java:58]: does it matter?

**Review by lajosrodek — APPROVED** [review]
LGTM

**Pull request merged**